java-checks/src/main/java/org/sonar/java/checks/CognitiveComplexityMethodCheck.java

```diff
@@ -116,10 +118,12 @@ private void increaseComplexityByOne(Tree tree) {
     private void increaseComplexity(Tree tree, int increase) {
       complexity += increase;
       if(!ignored.contains(tree)) {
```

Comment:
It won't work, as it will then report nothing for that `if` tree, while we still have to report the `+1`, but without the nesting penalty.